From 05e5405651054c580b248c4ab2791ed8d66369e3 Mon Sep 17 00:00:00 2001
From: Lukasz Czechowski <lukasz.czechowski@thaumatec.com>
Date: Mon, 3 Jun 2024 13:39:27 +0200
Subject: [PATCH] vnc: Allow neatvnc in version 0.8.0

Neat VNC 0.8.0 does not introduce any changes that breaks API used
by VNC backend, so it is safe to extend compatibility.

Signed-off-by: Lukasz Czechowski <lukasz.czechowski@thaumatec.com>
---
 kiosk-shell/kiosk-shell.c         | 5 +++--
 libweston/backend-vnc/meson.build | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/kiosk-shell/kiosk-shell.c b/kiosk-shell/kiosk-shell.c
index 28805322..a18580bc 100644
--- a/kiosk-shell/kiosk-shell.c
+++ b/kiosk-shell/kiosk-shell.c
@@ -880,8 +880,9 @@ desktop_surface_removed(struct weston_desktop_surface *desktop_surface,
 						     WESTON_ACTIVATE_FLAG_NONE);
 		} else {
 			kiosk_seat->focused_surface = NULL;
-			kiosk_shell_output_set_active_surface_tree(shoutput,
-								   NULL);
+			if (shoutput)
+				kiosk_shell_output_set_active_surface_tree(shoutput,
+									   NULL);
 		}
 	}

diff --git a/libweston/backend-vnc/meson.build b/libweston/backend-vnc/meson.build
index b7b69165..39b15cf9 100644
--- a/libweston/backend-vnc/meson.build
+++ b/libweston/backend-vnc/meson.build
@@ -3,7 +3,7 @@ if not get_option('backend-vnc')
 endif

 config_h.set('BUILD_VNC_COMPOSITOR', '1')
-dep_neatvnc = dependency('neatvnc', version: ['>= 0.7.0', '< 0.8.0'], required: false, fallback: ['neatvnc', 'neatvnc_dep'])
+dep_neatvnc = dependency('neatvnc', version: ['>= 0.7.0', '< 0.9.0'], required: false, fallback: ['neatvnc', 'neatvnc_dep'])
 if not dep_neatvnc.found()
 	error('VNC backend requires neatvnc which was not found. Or, you can use \'-Dbackend-vnc=false\'.')
 endif
--
